### ISSUE TITLE
Cannot create/reuse an activity when one with the same label has been deleted

### ISSUE THREAD

**Issue by nicobo** [ramack/ActivityDiary#109]
1. Create activity "my activity"
1. Delete it
1. Change your mind and create it again
1. Blocking message saying it already exists but cannot reuse the old one or create a new one

I've tricked the app by adding spaces in the label but it's not consistent since I would like history to relate to only one activity in this case.

**Comment by ramack** [comment]
Thanks for reporting. I was aware of that, but did even document it in an issue - shame on me. As a you faced that now in a real use case I will push this to the next release.

The plan is to add two possibilities:
- [ ] in the (currently not very nice) actitivity manager: add a checkbox to show also deleted activities and allow undeleting from there
- [ ] on creating a new activity (where you faced it), allow to undelete when you add an used name; here it might even be interesting to detect very similar names (e. g. your space trick) and ask whether this is really intended, or whether the similar name was intended

**Comment by ramack** [comment]
@iqbalridha do you want to propose icons for the actions
"undelete and edit the deleted activity with this name"
"rename the existing, deleted activity and go on editing the new one"
"an activity with that name already exists, edit that one"
those are to be shown beside the name textview in the edit activity dialog, in case of naming conflicts

"show deleted" / "hide deleted"
those should be in the activity manager to show and hide deleted activities.

Notes:
- your proposals must be published under a license compatible with the GPL v3+ (v3 or later).
- they should fit into the existing style of the app

**Comment by iqbalridha** [comment]
Well sir, i will try, make icon as per requirement. When finished I will confirm with you back

**Comment by iqbalridha** [comment]
This is the icon I have designed according to what you need pack. Icon I created a line-based. If there is anything you need you can contact me. How do you respond sir.
![icon-diary-activity](https://user-images.githubusercontent.com/36046668/37983449-b64449f0-321d-11e8-9cc2-953ea6841a7a.jpg)

**Comment by ramack** [comment]
Thanks for your proposal and the effort you put into this.
You might have seen the icons that I have sketched so far in the repository for the first 3 actions. - If not here are screenshots showing the use cases:
<img src="https://user-images.githubusercontent.com/153643/37985530-43ec94a8-31f9-11e8-97f4-f7fed7758382.png" width="280"> <img src="https://user-images.githubusercontent.com/153643/37985532-44093b4e-31f9-11e8-85dc-e0dbc6a3f1d4.png" width="280">

In 1 I focused on the trash to get it out. In your proposal I think the pen is too complex compared to the usual "android" edit pen and the getting it out of the trash is not really there. While in my first draft the edit of the "undisposed" item is not taken into account at all.

For your 2nd I do not visually understand a meaning of it.

3: I really like your idea of having the edit pen together with a "accept" check sign. Do you see a way to make it more in the style of the other android material icons? - The pen would there be not vertical, but rotated cw, but for sure this conflicts with the check. Do you have other ideas to overcome this? - vertically somehow I fear that the users do not directly link it to a pen...

4: There is some "movement" in your proposal, which I cannot really classify. Maybe we can do a combination of the trash bin and a magnifying glass to indicate a view into the trash, based on the material design icons? For the 4th case it might even be good to have two icons. one to show the deleted activities and the other to hide them again. I think I will show the deleted activities with the name in strikeout font effect or compeltely hide them - depending on the toggle state.

**Comment by iqbalridha** [comment]
No problem sir, I will fix this and come up with the idea and the fixed design

**Comment by iqbalridha** [comment]
have tried to improve and improvise it. Here's the result, how do you think sir.
![icon-diary-activity-re](https://user-images.githubusercontent.com/36046668/38047246-1b6d7f12-32ec-11e8-81a8-89bba3674289.jpg)

**Comment by ramack** [comment]
Thanks for your proposals. Honestly, I am grateful for you contributions, but I think your icons do not fit well into the style of the existing icons. I have added two more for the show and hide deleted in the "activity manager". See them in https://github.com/ramack/ActivityDiary/tree/master/artwork/icons For sure, finally it is a matter of taste what looks better and which style we should keep - similar to the existing one or change in the direction that you suggested.

We shall think of implementing the newly creates #128 to support different styles in the app. As I am an engineer and rate functionality higher than design I assigned it so far to release 1.4, but am happy to integrate it earlier if someone submits a PR.
